config/routes.rb

```diff
@@ -0,0 +1,116 @@
+Rails.application.routes.draw do
+  get 'media/index' => 'media#index', as: 'media'
+  root 'media#index'
+
+
+patch 'books/:id/upvote' => 'books#upvote'
+patch 'movies/:id/upvote' => 'movies#upvote'
+patch 'albums/:id/upvote' => 'albums#upvote'
+
+patch 'books/:id/downvote' => 'books#downvote'
+patch 'movies/:id/downvote' => 'movies#downvote'
+patch 'albums/:id/downvote' => 'albums#downvote'
```

Comment:
These custom actions can actually be included directly with the `resources` lines below:
```ruby
resources :albums do
  member do
    patch 'upvote'
    patch 'downvote'
  end
end
```